src/CommandIsolator.cpp

```diff
@@ -82,6 +90,40 @@ process::Future<Option<ContainerLaunchInfo>> CommandIsolatorProcess::prepare(
   return containerLaunchInfo.get();
 }
 
+process::Future<ContainerLimitation> CommandIsolatorProcess::watch(
```

Comment:
Why don't you use an optional as the prepare command does ?

Reply:
`None()` would be considered as something, so it won't match [the default constructor](https://github.com/criteo-forks/mesos/blob/criteo/1.7.0/3rdparty/libprocess/include/process/future.hpp#L1063-L1068) which is the only one setting the `Future` as *abandoned*.
The *abandoned* state is important because it is the only one which won't trigger a limitation from the containerizer (neither killing the container nor reporting a failure).